### PR TITLE
[Woo POS] Simplify `TotalsViewModel` protocol. Update access control

### PR DESCRIPTION
**Pull request by iamgabrielma**
## Description
This PR attempts to simplify the existing `TotalsViewModelProtocol` and its implementations. By making private several properties and functions that were never accessed from the outside, we can make the design simpler. 

I left some comments below as there's still room for improvement, but this looks like a good first step.

## Changes
- Deleted empty placeholder tests from the initial states of the POC, most of them are no longer valid.
- Made properties and methods private when not being accessed from outside the class, this lead to deleting their signature from the protocol, and the associated mock implementation (if any).
- Moved currency formatter helpers to a private extension.

## Testing information

CI should pass. 

Consider smoke testing the payment flow, and the totals view, but there are no changes to the business logic: The computed properties that have been deleted from the view model weren't used anywhere except for protocol conformance, and the publicly-exposed properties that we've restricted remain gettable via `private(set)`.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/224">20.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift [hunk 1]
```diff
@@ -34,27 +34,15 @@ final class TotalsViewModel: ObservableObject, TotalsViewModelProtocol {
 
     @Published private(set) var connectionStatus: CardReaderConnectionStatus = .disconnected
 
-    @Published var formattedCartTotalPrice: String?
-    @Published var formattedOrderTotalPrice: String?
-    @Published var formattedOrderTotalTaxPrice: String?
+    @Published private(set) var formattedCartTotalPrice: String?
+    @Published private(set) var formattedOrderTotalPrice: String?
+    @Published private(set) var formattedOrderTotalTaxPrice: String?
```
Comment (iamgabrielma) [inline WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift:39]: I don't think we can properly test these in the current state, so I left it for a future PR:
- Populating values for these properties its done via `TotalsViewModel.updateOrder()` which is private, and is invoked as a side-effect of `TotalsViewModel.syncOrder()`. It seems an indication that this also should be moved somewhere else, perhaps some sort of `OrderUpdater` or a small struct that contains all the transaction data, then is passed into the view model.
- Even if we could test the invocation of `updateOrder()`,  the result would tell us if the `CurrencyFormatter` works as expected, which is tested extensively on its own, and shouldn't be part of this.

**Comment by iamgabrielma** [comment]
No rush for the review! I've set it to 20.6 for now, but can be done after HACK week as well 👍

**Review by staskus — APPROVED** [review]
Thanks for cleaning it up 👍 It will help us with further refactoring.

All of this is only here because we want to reuse and inject view models to one another. If we stop doing that, then we can make even further clean up.

**Pull request merged**